internal/services/tesla_fleet_api_service.go

```diff
@@ -164,12 +178,81 @@ func (t *teslaFleetAPIService) GetAvailableCommands() *UserDeviceAPIIntegrations
 	}
 }
 
-// performTeslaGetRequest a helper function for making http requests, it adds a timeout context and parses error response
-func (t *teslaFleetAPIService) performTeslaGetRequest(ctx context.Context, url, token string) (*http.Response, error) {
+// VirtualTokenConnectionStatus Checks whether vehicles can accept Tesla commands protocol for the partner's public key
+func (t *teslaFleetAPIService) VirtualTokenConnectionStatus(ctx context.Context, token, region, vin string) (bool, error) {
+	baseURL := fmt.Sprintf(t.Settings.TeslaFleetURL, region)
+	url := fmt.Sprintf("%s/api/1/vehicles/fleet_status", baseURL)
+
+	jsonBody := []byte(fmt.Sprintf(`{"vins": ["%s"]}`, vin))
```

Comment:
If we're going to do string manipulation,`fmt.Sprintf(`{"vins": [%q]}`, vin)` feels a little better to me. Minor point.